app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/repository/RepositoryCommitViewModel.kt

```diff
@@ -0,0 +1,71 @@
+package com.bottlerocketstudios.brarchitecture.ui.repository
+
+import androidx.annotation.VisibleForTesting
+import com.bottlerocketstudios.brarchitecture.R
+import com.bottlerocketstudios.brarchitecture.data.model.CommitDto
+import com.bottlerocketstudios.brarchitecture.data.repository.BitbucketRepository
+import com.bottlerocketstudios.brarchitecture.domain.models.Status
+import com.bottlerocketstudios.brarchitecture.infrastructure.util.exhaustive
+import com.bottlerocketstudios.brarchitecture.ui.BaseViewModel
+import com.bottlerocketstudios.compose.repository.RepositoryCommitItemUiModel
+import com.bottlerocketstudios.compose.util.formattedUpdateTime
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.map
+import org.koin.core.component.inject
+import java.time.Clock
+
+class RepositoryCommitViewModel : BaseViewModel() {
+    // DI
+    private val repo: BitbucketRepository by inject()
+    private val clock by inject<Clock>()
+
+    companion object {
+        const val HASH_LENGTH = 6
+    }
+
+    // State
+    private val srcCommits = MutableStateFlow<List<CommitDto>>(emptyList())
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    var currentRepoName: String = ""
```

Comment:
Could we make this a mutable state flow?  Then:

init {
  launchIO {
    currentRepoName.collect { repoName -> getRepoCommits(repoName)
}
}

Reply:
This will allow us to make getRepoCommits private and then anytime currentRepoName changes value it will fetch the correct info.

Reply:
Got it! I sloppily wrote this down in my notes. I'll get this done!